### PR TITLE
travis

### PR DESCRIPTION
**Pull request by SylwiaGargula**
Travis and rpm.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,29 @@
...
+env:
+  global:
+    - secure: "P+mNX77uTNYI/xpI+2w+y1+d6w81ovEVxpB8mp4QTb0CTJGc+Ms1Mqp4ja3O75wiPRsaEYHQOveehXcVlQPWn10m95K/HRxFeBIvBb4DiugHq53gchbjPADDVjt7fCh5x9YNVH5Bho8wuDbG966vfFnTS1k0BjpHOOt6GK7lquVTbjonKF/V8BvdYTlhlajN158dj6kw2AiWguNdswZNM1DNI3YILZSGJqkXn3BVGTBUiKKLB6FI+uJJDpr+giktytU1Xx+HllMN30rICvI4Y7Ro3u/ZAs6/vMVERvfj31N2uibfCoPlyFozEPv/iBevlCqL/QxltiAj04eMRW4nMHk/Ey0ZyWbBc7dgCZ3f+Kej6/HKmLdVel3LFCBYH31q9D21S/6kpwQkMYSyzWdwWYdB8CBTKGv/szbMMo7QhCMN54WBy5VldiBlrtealP9UJNxSUbb8MG3qsuuxAaeqVcmmY1AWiPJtJEHh5HmE6dPsOzZK5qn/8U3PWsn3lYhUtiivBXnH3reiY7UT55eqyb5vT6AWoFrjyiPq7DjYa/o5hQEiAgKZV8Mxh8WQoSkG1uoK+ipE//5n8UkK2VV6xhilcPavBvfbKDNs1jKRgFUyG7TvAVVl+uC51BP2zGgsy88DGH7xlz3sWukpX7rIXwqD/1RxlfsK9cIa3nDdwBM="
+    - git: "b946ea1a81da66a6bd121d4bdfc3b97357d996d5"
 install: make config-install get-deps
 
 script:
   - make code-analysis
   - make test
-
```
Comment (pfigwer) [inline .travis.yml]: Why did you get rid of all deployment steps?

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,29 @@
...
-  file:
-    - release/perun-linux-amd64.tar.gz
-    - release/perun-darwin-amd64.tar.gz
-    - release/perun-windows-amd64.tar.gz
-    - release/perun-windows-386.tar.gz
-  skip_cleanup: true
-  on:
-    tags: true
+	
+after_success:
```
Comment (pfigwer) [inline .travis.yml]: `after_success` will be performed after every succeded build. I think what you want to use here is `after_deploy` step.

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,29 @@
...
-    secure: eLTZravNiDVvQ1dekb7NvWYJBIG2X6CzfHEqUKJ2JCDlKSqYXrvqZF/B3XxmYvmF1tEAmAa08LMDzzzApl9IML1DVSoW8i6uy+uetg+xbvumAf9fq14nMd0JQEEA1qruE7pwjyQs7h9gXYtyAR01CPhj/xNUQmYV1i8NCTHoljBkO+NsMFyi3WMbW7HTRQZQZXbPBagI06L3tSOCfN/w5KVmRsKFQ3lvmnzs+mTrIvOy2CBQC+0Cp3PQ/p7yyhEWRFd5J6n2jYGxneetnBq0FAfbOF4RIwvrWuu9XI/znxhYMOB5lra0qUwuG+prJStB6oaQ/vHStRcxQorV75Jtm4u/EHcFmmaxTQvPksdZQ8VSIbFonz1qbnuurP5sloiAR1RnJQtQWZKj7I7ioknEBh4kqCGvLUIbt0VpHTNoPKN0a8GYiPSE9UO6J+CNS+FR5mahW3xsHx5dHMV+R4mxcbt16dlg0g8m4tah06bd3P/t91kkgliTWmkHDMX4ES4hh+ribMnsLB0k7iqtuoO2P+gFn80CR5ooAX9Z3u8P8MaEovuPSaO7DqsGfX3uCaFInyBpc5EteCNwgN9dGAfh4mscJlijx28qgJ5quNU56fhcfQ8DoC5nXTM7RRRSu0OB1xSDa9OEf5Nh1AlkDwQKxjAYD+ujYFCXxqSWcntUbqE=
-  file:
-    - release/perun-linux-amd64.tar.gz
-    - release/perun-darwin-amd64.tar.gz
-    - release/perun-windows-amd64.tar.gz
-    - release/perun-windows-386.tar.gz
-  skip_cleanup: true
-  on:
-    tags: true
+
```
Comment (afronski) [inline .travis.yml]: When it comes to `Makefile` it is better to remove the obsolete whitespace.

File: TravisTest.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+1
```
Comment (afronski) [inline TravisTest.txt]: Is that file needed?

File: .travis.yml [hunk 5]
```diff
@@ -1,34 +1,29 @@
...
+  - 'wget -P SOURCES https://raw.githubusercontent.com/Appliscale/perun/master/defaults/main.yaml'
+  - 'wget -p SOURCES https://raw.githubusercontent.com/Appliscale/perun/master/LICENSE'
+  - 'tar xvzf perun-linux-amd64.tar.gz'
+  - 'rm perun-linux-amd64.tar.gz'
+  - 'tar cvzf perun-linux-amd64.tar.gz perun-linux-amd64 LICENSE main.yaml'
+  - 'rm LICENSE main.yaml perun-linux-amd64'
+  - 'cd ..'
+  - 'rpmbuild -ba /SPEC/perun-linux-amd64.spec'
+  - 'git add .'
+  - 'git commit -m "update rpm by travis"'
```
Comment (afronski) [inline .travis.yml]: A small improvement would be to prettify that message e.g..
```
"[AUTO] Update RPM by Travis CI."
```

File: .travis.yml [hunk 6]
```diff
@@ -32,3 +37,20 @@ deploy:
   skip_cleanup: true
   on:
     tags: true
+
+after_deploy:
+  - 'git clone https://github.com/Appliscale/rpmbuild.git'
+  - 'cd rpmbuild/SOURCES'
+  - 'rm perun-linux-amd64.tar.gz'
+  - 'wget -P SOURCES https://github.com/Appliscale/perun/releases/download/1.2.0/perun-linux-amd64.tar.gz'
```
Comment (pfigwer) [inline .travis.yml]: We want to download binary from tag we are curently releasing, not hardcoded `1.2.0`. You can get it from `TRAVIS_TAG` envirnoment variable.

File: .travis.yml [hunk 7]
```diff
@@ -32,3 +32,39 @@ deploy:
...
+  - 'wget https://raw.githubusercontent.com/Appliscale/perun/master/defaults/main.yaml'
+  - 'wget https://raw.githubusercontent.com/Appliscale/perun/master/LICENSE'
+  - 'tar xvzf perun-linux-amd64.tar.gz'
+  - 'rm perun-linux-amd64.tar.gz'
+  - 'tar cvzf perun-linux-amd64.tar.gz perun-linux-amd64 LICENSE main.yaml'
+  - 'rm LICENSE main.yaml perun-linux-amd64'
+  - 'cd ..'
+  - 'rpmbuild -ba /SPEC/perun-linux-amd64.spec'
+  - 'git remote'
+  - 'git config user.email sylwia.gargula@wp.pl'
```
Comment (pfigwer) [inline .travis.yml]: I would create Travis credentials here.

File: .travis.yml [hunk 8]
```diff
@@ -32,3 +32,39 @@ deploy:
   skip_cleanup: true
   on:
     tags: true
+
+after_deploy:
+  - 'git clone https://github.com/Appliscale/rpmbuild.git'
```
Comment (pfigwer) [inline .travis.yml]: Maybe we can create a script, e.g. `release.sh`, and just call it here? With so many steps travis config is starting to looks unclear.

File: .travis.yml [hunk 9]
```diff
@@ -1,8 +1,14 @@
 language: go
 
+sudo: required
```
Comment (pfigwer) [inline .travis.yml]: Do we really need sudo here?

File: .travis.yml [hunk 10]
```diff
@@ -1,8 +1,14 @@
...
+
 go:
   - 1.9.x
 
+env:
+  global:
+    - secure: "roiMALXQRkKKlPP+cbtac795XeHHnPrKvc6M8HjVa5gzL1SZXHDn/zvcvTc//AAMBFxZXb9+V+USrU3vNyzULIoh5j7nTOutPtik4WYLIfwADVgSxpiEG9jDBYp3DLlXMAMb5uJWRuG4DSSxpUGPFujQr9+aaUa0Lz6tsPUpE2/yaQfR5GmrAVteevkEZ0h18BxiRKsf4cRIGEQpwOSAQ+x66MlaEAn/kkprS4K9imniHeDRnz4TM75SAI7VZh9Fglv6gf8HTZ73EV1sGzsbG8+k1eO53UxMYR7YneB86ihKjRFuXSXzT8K0RdoACY6Muf81kwP5JUP3SLMaNgRTrdGmzOZ+yhJzfuey0b7i2llvGHrfKCmEIUzTYC5KZgvwblqwU97u55baJpAwQJgwPOKBO2RUvyzdK00t39kj+PpGrmyrW9evQ7mGk7o7txn4K1LJfCTn/fll1Y3B1poyiflYfQ4vlj7RgKSwAgIs36+yPhYF/m1K9I+jy8BRHPCZek3JJIMnKaW3OEBzpCRBoxxmMFomN8tq2Ygde5TUVxukAbZNIrlaPPWB5ZB6ZQ1hKol0yddkqOKxivGPQJaZthlt7eHpx4rAG+ZY1e706pG3iqYcTEVJTjPPzCaKdSpbFg3oKGet3ClimF0ycNpJRFQs2eveI3AkhVCN2Z4IWX8=" #Github
```
Comment (pfigwer) [inline .travis.yml]: Do we need another key here? Can't we use the same one we are using in deploy step?

**Comment by pfigwer** [comment]
I think we don' need `.release.sh.swp` file.

File: .travis.yml (hunk 9 above)
Comment (SylwiaGargula) [reply .travis.yml]: I need sudo to install rpm.

File: release.sh [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+rm $perun.tar.gz
+tar cvzf $perun.tar.gz $perun LICENSE main.yaml
+rm LICENSE main.yaml $perun
+cd ..
+rpmbuild -ba SPECS/$perun.spec
+git remote
+git config user.email $email
+git config user.name $user
+git add .
+git commit -m "[AUTO] Update RPM by Travis CI"
```
Comment (pfigwer) [inline release.sh]: Add Perun version to commit messages.
